acme/challenge.go

```diff
@@ -303,21 +349,32 @@ func newHTTP01Challenge(db nosql.DB, ops ChallengeOptions) (challenge, error) {
 // updated.
 func (hc *http01Challenge) validate(db nosql.DB, jwk *jose.JSONWebKey, vo validateOptions) (challenge, error) {
 	// If already valid or invalid then return without performing validation.
-	if hc.getStatus() == StatusValid || hc.getStatus() == StatusInvalid {
+	if hc.getStatus() == StatusValid {
+		return hc, nil
+	}
+	if hc.getStatus() == StatusInvalid {
+		// TODO: Resolve segfault on upd.save
```

Comment:
This happens not as a result of the new test in this PR, but because things in the invalid state now hit the db, and the "ok/status-already-invalid" test doesn't provide a mock db so the test runner tries to invoke a null function.